pkg/applications/providers/util/util.go

```diff
@@ -44,3 +50,49 @@ func CleanUpHelmTempDir(cacheDir string, logger *zap.SugaredLogger) {
 
 // StatusUpdater is a function that postpone the update of the applicationInstallation.
 type StatusUpdater func(status *appskubermaticv1.ApplicationInstallationStatus)
+
+// GetCredentialFromSecret get the secret and returns secret.Data[key].
+func GetCredentialFromSecret(ctx context.Context, client ctrlruntimeclient.Client, namespce string, name string, key string) (string, error) {
+	secret := &corev1.Secret{}
+	if err := client.Get(ctx, types.NamespacedName{Namespace: namespce, Name: name}, secret); err != nil {
+		return "", fmt.Errorf("failed to get credential secret: %w", err)
+	}
+
+	cred, found := secret.Data[key]
+	if !found {
+		return "", fmt.Errorf("key '%s' does not exist in secret '%s'", key, fmt.Sprintf("%s/%s", secret.GetNamespace(), secret.GetName()))
+	}
+	return string(cred), nil
+}
+
+// AuthFromCredentials builds helmclient.AuthSettings from source.Credentials. registryConfigFilePath is the path of the file that stores credentials for OCI registry.
+func AuthFromCredentials(ctx context.Context, client ctrlruntimeclient.Client, registryConfigFilePath string, secretNamespace string, source *appskubermaticv1.HelmSource) (helmclient.AuthSettings, error) {
```

Comment:
tiniest nitpick (just because I have nothing else to complain about, because everything's so nice :D)

This could inlcude Helm in the name of the func, since it is in a generic package called util and it does helm specific things